core/src/main/scala/org/apache/spark/SparkException.scala

```diff
@@ -71,9 +71,35 @@ private[spark] case class ExecutorDeadException(message: String)
 /**
  * Exception thrown when Spark returns different result after upgrading to a new version.
  */
-private[spark] class SparkUpgradeException(version: String, message: String, cause: Throwable)
-  extends RuntimeException("You may get a different result due to the upgrading of Spark" +
-    s" $version: $message", cause)
+private[spark] class SparkUpgradeException(
+    version: String,
+    message: String,
+    cause: Throwable,
+    errorClass: Option[String],
+    messageParameters: Array[String])
+  extends RuntimeException(message, cause) with SparkThrowable {
+
+  def this(version: String, message: String, cause: Throwable) =
+    this (
+      version = version,
+      message = s"You may get a different result due to the upgrading of Spark $version: $message",
```

Comment:
Let's avoid error message duplication. error-classes.json should be one source of truth:
```
  "INCONSISTENT_BEHAVIOR_CROSS_VERSION" : {
    "message" : [ "You may get a different result due to the upgrading of Spark %s: %s" ]
  },
```

Reply:
Thanks @MaxGekk. Will fix this.